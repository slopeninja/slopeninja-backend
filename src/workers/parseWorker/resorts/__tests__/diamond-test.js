import fs from 'fs';
import {
  parseDiamondSnow,
  parseDiamondLiftCounts,
  parseDiamondLifts,
  parseDiamondTrails,
  parseDiamondTrailCounts,
} from '../diamond';
import { createHtmlParser } from '../../parserFactory';

test('fetches Diamond snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/diamond-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseDiamondSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: 6,
      snowDepthBase: 33,
      snowDepthSummit: 66,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseDiamondSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Diamond lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/diamond-weather.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseDiamondLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseDiamondLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Diamond trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/diamond-weather.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseDiamondTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseDiamondTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Diamond lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/diamond-weather.html.input`);
  const resortData = await createHtmlParser('lifts', parseDiamondLifts)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('lifts', parseDiamondLifts)('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Diamond trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/diamond-weather.html.input`);
  const resortData = await createHtmlParser('trails', parseDiamondTrails)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trails', parseDiamondTrails)('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});
