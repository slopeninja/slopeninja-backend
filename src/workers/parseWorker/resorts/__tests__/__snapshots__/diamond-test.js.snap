// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Diamond lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": null,
      "name": "Crystal Express - High-Speed Quad",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Lakeview Lift - Quad",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Lodgepole Lift - Quad",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Pete's Powerline",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Red Fox Lift - Double",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Ridge Chair - Double",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Schoolhouse Lift - Double",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Terrain Parks",
      "status": null,
    },
  ],
}
`;

exports[`fetches Diamond trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Battle Born",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Crystal Ridge",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Diamondback",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Golden Eagle Bowl",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Great Flume",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Lakeview",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Lightning",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Luggi's",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Powder",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Spillway",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Sunnyside",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Thunder",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Area del Fuego",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Boarder Line",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Bonanza",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Elevator Shaft",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Glade Runner",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Gnarnia",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Nirvana",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Outer Limits",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Promised Land",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Rollerglades",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Solitude Canyon",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "The Crown Jewels",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Chute",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Corkscrew",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Dusty's",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "F.I.S.",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "beginner",
      "name": "Freeway",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "G.S.",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Lower Show Off",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "O God",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Penguin",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Popular",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Ridge Run",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Slalom Glade",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Upper Show Off",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "intermediate",
      "name": "Wiggle",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Cappy's Corner",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "advanced",
      "name": "Zig Zag",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "beginner",
      "name": "Lodgepole",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": "beginner",
      "name": "School Yard",
      "status": "open",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": null,
      "name": "The Village - Jump Start",
      "status": "closed",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": null,
      "name": "The Village - Spillway",
      "status": "closed",
    },
    Object {
      "category": "Crystal Express - High-Speed Quad",
      "level": null,
      "name": "The Village - Wiggle",
      "status": "open",
    },
  ],
}
`;
